src/Symfony/Component/String/Utf8String.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * Represents a string of Unicode code points encoded as UTF-8.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+class Utf8String extends AbstractUnicodeString
+{
+    public function __construct(string $string = '')
+    {
+        if ('' !== $string && !preg_match('//u', $string)) {
+            throw new InvalidArgumentException('Invalid UTF-8 string.');
+        }
+
+        $this->string = $string;
+    }
+
+    public function append(string ...$suffix): AbstractString
+    {
+        $str = clone $this;
+        $str->string .= 1 >= \count($suffix) ? $suffix[0] ?? '' : implode('', $suffix);
```

Comment:
I've added parentheses around the coalescing operation. About the general CS, please send PRs after merge. To my eyes, the code is as readable as I wish it could be :)